mmpose/datasets/pipelines/gesture_transform.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import mmcv
+import numpy as np
+import torch
+
+from mmpose.core import bbox_xywh2xyxy, bbox_xyxy2xywh
+from mmpose.datasets.builder import PIPELINES
+
+
+@PIPELINES.register_module()
+class CropValidClip:
+    """Generate the clip from complete video with valid frames.
+
+    Required keys: 'video', 'modality', 'valid_frames', 'num_frames'.
+
+    Modified keys: 'video', 'valid_frames', 'num_frames'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Crop the valid part from the video."""
+        if 'valid_frames' not in results:
+            results['valid_frames'] = [[0, n - 1]
+                                       for n in results['num_frames']]
+        lengths = [(end - start) for start, end in results['valid_frames']]
+        length = min(lengths)
+        for i, modal in enumerate(results['modality']):
+            start = results['valid_frames'][i][0]
+            results['video'][i] = results['video'][i][start:start + length]
+            results['num_frames'] = length
+        del results['valid_frames']
+        if 'bbox' in results:
+            results['bbox'] = results['bbox'][start:start + length]
+        return results
+
+
+@PIPELINES.register_module()
+class TemporalPooling:
+    """Pick frames according to either stride or reference fps.
```

Comment:
Does this mean stride and reference fps should not be set simultaneously?

Reply:
Yes, it seems better to add an assertion here.